scripts/firedrake-configure

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/env python3
+
+"""Script for preparing an environment to install Firedrake into."""
+
+# To avoid the pitfalls inherent in having executable configuration files
+# this script is intentionally extremely dumb. All configure options are computed
+# 'statically' (at import) and the only run-time logic that happens is dispatching
+# on the right package manager and arch.
+
+# As a matter of policy, new package managers and archs should only be added to
+# this file if they are *tested in CI*.
+
+import argparse
+import enum
+import os
+import platform
+import subprocess
+from collections.abc import Sequence
+from typing import Optional
+
+
+class PackageManager(enum.Enum):
+    LINUX_APT_X86_64 = "apt-x86_64"
+    LINUX_APT_AARCH64 = "apt-aarch64"
+    MACOS_HOMEBREW_ARM64 = "brew-arm64"
+
+
+LINUX_APT_X86_64 = PackageManager.LINUX_APT_X86_64
+LINUX_APT_AARCH64 = PackageManager.LINUX_APT_AARCH64
+MACOS_HOMEBREW_ARM64 = PackageManager.MACOS_HOMEBREW_ARM64
+
+
+class FiredrakeArch(enum.Enum):
+    DEFAULT = "default"
+    COMPLEX = "complex"
+
+
+ARCH_DEFAULT = FiredrakeArch.DEFAULT
+ARCH_COMPLEX = FiredrakeArch.COMPLEX
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Print out configuration options needed to install Firedrake",
```

Comment:
Bit more detail and reference back to website.

Reply:
done